src/ucp/core/ucp_worker.c

```diff
@@ -2339,17 +2339,11 @@ static void ucp_worker_destroy_eps(ucp_worker_h worker)
     ucs_debug("worker %p: destroy all endpoints", worker);
     ucs_list_for_each_safe(ep_ext, tmp, &worker->all_eps, ep_list) {
         ep = ucp_ep_from_ext_gen(ep_ext);
-        if (ucp_ep_config(ep)->key.err_mode == UCP_ERR_HANDLING_MODE_PEER) {
-            /* Discard all lanes to purge pending and outstanding operations to
-             * not purge pending requests with ucp_destroyed_ep_pending_purge()
-             * callback in ucp_ep_cleanup_lanes() function when destroying the
-             * UCP EP below.
-             * Do discarding of lanes if the UCP EP was created with a peer
-             * failure detection support requested by a user, because it does
-             * flush(CANCEL) which can be used only for transports which support
-             * a peer failure detection feature */
-            ucp_ep_discard_lanes(ep, UCS_ERR_CANCELED);
-        }
+        /* Purge pending operations scheduled on this UCP EP to not do this with
+         * ucp_destroyed_ep_pending_purge() callback in ucp_ep_cleanup_lanes()
+         * function when destroying the UCP EP below */
```

Comment:
what is the problem with doing it from ucp_destroyed_ep_pending_purge?
maybe we should replace ucp_destroyed_ep_pending_purge by ucp_ep_err_pending_purge?

Reply:
`ucp_destroyed_ep_pending_purge()` was added to report that a user's requests weren't sent correctly, but EP was closed
so, removing this, we won't be able to catch this anymore

Reply:
so should it be like
```
/* Cleanup pending operations on the endpoint, since ucp_ep_destroy_internal() expects the pending
queues of the UCT endpoints will be empty before they are destroyed */
```

Reply:
yes

Reply:
done